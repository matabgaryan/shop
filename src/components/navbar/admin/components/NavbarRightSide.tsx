import React from "react";
import {logout} from "../../../../store/slices/auth";
import {useNavigationHook} from "../../../../hooks/useNavigation";
import {ROUTES} from "../../../../configs/routes";
import {useAppDispatch} from "../../../../hooks/useRedux";

export const NavbarRightSide = () => {
    const dispatch = useAppDispatch();
    const { handleNavigate } = useNavigationHook();

    const handleLogout = () => {
        dispatch(logout());
        handleNavigate(ROUTES.adminLogin)
    }

    return(
        <div className="hidden md:flex items-center justify-end md:flex-1 lg:w-0">
                    <button
                        onClick={() => handleLogout()}
                        className="h-12 w-40	 whitespace-nowrap inline-flex items-center justify-center px-4 py-2 border border-appBlack rounded-md shadow-sm text-[14] font-[400] text-oziBlack bg-transparent hover:opacity-80"
                    >
                        Logout
                    </button>
        </div>

    )
}
