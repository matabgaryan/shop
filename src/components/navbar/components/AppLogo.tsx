import React from "react";
import {Link} from "react-router-dom";
import AppIcon from "../../../assets/icons/appLogo.png";

export const AppLogo: React.FC = () => {
    return(
        <div className="flex justify-start ">
            <Link to="/">
                <img
                    className="h-8 w-auto sm:h-10"
                    src={AppIcon}
                    alt=""
                />
            </Link>
        </div>
    )
}
