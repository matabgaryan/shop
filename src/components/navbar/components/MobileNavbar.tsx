import React from "react";
import AppIcon from "../../../assets/icons/appLogo.png";
import {Link} from "react-router-dom";
import {IMobileNavbarProps} from "../interface";
import {CloseIcon} from "../../icons/CloseIcon";

export const MobileNavbar: React.FC<IMobileNavbarProps> = ({open, setOpen, routes}) => {

    return(
        <div
            className={
                open
                    ? "opacity-100 z-1024 scale-100 transition ease-out duration-200 absolute top-0 inset-x-0 p-2 transition transform origin-top-right md:hidden"
                    : "opacity-0 scale-95 absolute top-0 inset-x-0 p-2 transition transform origin-top-right md:hidden"
            }
        >
            <div className="rounded-lg shadow-lg ring-1 ring-appBlack ring-opacity-5 bg-white divide-y-2 divide-gray-50">
                <div className="pt-5 pb-6 px-5">
                    <div className="flex items-center justify-between">
                        <div>
                            <img
                                className="h-8 w-auto"
                                src={AppIcon}
                                alt="Workflow"
                            />
                        </div>
                        <div className="-mr-2">
                            <button
                                type="button"
                                className="bg-white rounded-md p-2 inline-flex items-center justify-center text-gray-400 hover:text-gray-500 hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-indigo-500"
                                onClick={() => setOpen(!open)}
                            >
                                <span className="sr-only">Close menu</span>
                                <CloseIcon />
                            </button>
                        </div>
                    </div>
                    <div className="mt-6">
                        <nav className="grid gap-y-8">
                            {
                                routes.map((item) => {
                                    return (
                                        <Link
                                            key={item.path}
                                            to={item.path}
                                            className="-m-3 p-3 flex items-center rounded-md hover:bg-gray-50"
                                        >
                                        <span className="ml-3 text-base font-medium text-gray-900">
                                          {item.name}
                                        </span>
                                        </Link>
                                    )
                                })
                            }
                        </nav>
                    </div>
                </div>
            </div>
        </div>
    )
}
