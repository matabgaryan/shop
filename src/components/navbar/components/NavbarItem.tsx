import React, {useEffect, useState} from "react";
import { NavLink, useLocation } from "react-router-dom";
import {INavItemProps} from "../interface";

export const NavbarItem: React.FC<INavItemProps> = ( {navItem}) => {
    const [activeRoute, setActiveRoute] = useState('/')
    const routeLocation = useLocation();

    useEffect(() => {
        setActiveRoute(routeLocation.pathname)
    },[routeLocation.pathname])

    return (
        <div
            className="flex flex-col justify-center items-center"
            key={navItem.path}>
            <NavLink
                key={navItem.path}
                onClick={() => setActiveRoute(navItem.path)}
                to={navItem.path}
                className={({ isActive }) =>
                    `font-GoogleSansRegular text-base font-[18px]  hover:text-gray-900 
                    ${isActive ? 'text-appBlack' : 'text-lightGrey'}`
                }
            >
                {navItem.name}
            </NavLink>
            {
                activeRoute === navItem.path &&
                <p className="w-[14px] h-[3px] bg-oziPurple"/>
            }
        </div>
    )
}
