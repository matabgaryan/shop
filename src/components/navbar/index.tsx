import AppNavBar from "./AppNavbar";

export default AppNavBar
