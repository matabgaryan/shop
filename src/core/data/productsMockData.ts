import {generatedUuid} from "../utils/uuid";
import {IProduct} from "../../store/slices/interface";

export const productsData: IProduct[] = [
    {
        id: generatedUuid(),
        name: 'Product 1',
        price: '30',
        weight: 100,
        startDate: 'May 5 , 2022',
        endDate: 'Jun 5 2022',
        count: 0
    },
    {
        id: generatedUuid(),
        name: 'Product 2',
        price: '30',
        weight: 100,
        startDate: 'May 5 , 2022',
        endDate: 'Jun 5 2022',
        count: 0
    },
    {
        id: generatedUuid(),
        name: 'Product 3',
        price: '30',
        weight: 20,
        startDate: 'May 5 , 2022',
        endDate: 'Jun 5 2022',
        count: 0
    },
    {
        id: generatedUuid(),
        name: 'Product 4',
        price: '30',
        weight: 30,
        startDate: 'May 5 , 2022',
        endDate: 'Jun 5 2022',
        count: 0
    },
    {
        id: generatedUuid(),
        name: 'Product 5',
        price: '30',
        weight: 40,
        startDate: 'May 5 , 2022',
        endDate: 'Jun 5 2022',
        count: 0
    },
    {
        id: generatedUuid(),
        name: 'Product 6',
        price: '30',
        weight: 200,
        startDate: 'May 5 , 2022',
        endDate: 'Jun 5 2022',
        count: 0
    },
    {
        id: generatedUuid(),
        name: 'Product 7',
        price: '30',
        weight: 50,
        startDate: 'May 5 , 2022',
        endDate: 'Jun 5 2022',
        count: 0
    },
    {
        id: generatedUuid(),
        name: 'Product 8',
        price: '30',
        weight: 100,
        startDate: 'May 5 , 2022',
        endDate: 'Jun 5 2022',
        count: 0
    },
    {
        id: generatedUuid(),
        name: 'Product 9',
        price: '30',
        weight: 65,
        startDate: 'May 5 , 2022',
        endDate: 'Jun 5 2022',
        count: 0
    },
    {
        id: generatedUuid(),
        name: 'Product 10',
        price: '30',
        weight: 70,
        startDate: 'May 5 , 2022',
        endDate: 'Jun 5 2022',
        count: 0
    },
]
