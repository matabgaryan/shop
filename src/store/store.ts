import {combineReducers, configureStore, PreloadedState} from "@reduxjs/toolkit";
import shopCart from "./slices/shopCart";
import productsReducer from "./slices/products";
import authReducer from './slices/auth';
import modalReducer from './slices/modals';
import {RootState} from "../core/types";

export const store = configureStore({
    reducer:{
        shopCart,
        products: productsReducer,
        auth: authReducer,
        modals: modalReducer,

    },
    devTools: true,
})
const rootReducer = combineReducers({
    shopCart,
    products: productsReducer,
    auth: authReducer,
    modals: modalReducer,
});


export function setupStore(preloadedState?: PreloadedState<RootState>) {
    return configureStore({
        reducer: rootReducer,
        preloadedState
    })
}

