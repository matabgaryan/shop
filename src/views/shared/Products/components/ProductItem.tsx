import React from "react";
import {AttributeInfo} from "../../../../components/attribute";
import ProductImage from '../../../../assets/images/productItem.jpeg';
import {IProduct} from "../../../../store/slices/interface";

interface IProductItemProps {
    product: IProduct,
    addProductToCart: (item: any) => void
}

export const ProductItem: React.FC<IProductItemProps> = ({product, addProductToCart}) => {
    return (
        <div
            data-testid="product-item"
            className="container mx-auto p-6 bg-white max-w-sm rounded-2xl overflow-hidden shadow-xl hover:shadow-2xl transition duration-300">
            <img className="rounded-xl" src={ProductImage} alt="Product"/>
            <div className="flex flex-col">
                <h1 data-testid="product-item-name" className="mt-5 text-lg font-semibold">{product.name}</h1>
                <AttributeInfo
                    attribute="Price"
                    value={`${product.price}$`} />
                <AttributeInfo
                    valueClassName="p-1 bg-oziRed text-white rounded"
                    attribute="Weight" value={`${product.weight} kg`} />
                <AttributeInfo
                    attribute="Start Date"
                    value="May 10, 2022" />
                <AttributeInfo
                    attribute="End Date"
                    value="May 10, 2022" />
            </div>
            <div className="mt-6 text-center">
                <button
                    onClick={() => addProductToCart(product)}
                    className="text-white text-sm	 font-semibold bg-appBlack py-1 px-2 rounded-md shadow-md  transition duration-500 transform-gpu  ">
                    Add To Card
                </button>
            </div>
        </div>
    )
}
