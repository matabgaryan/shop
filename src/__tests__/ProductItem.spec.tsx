import {
    render,
    screen,
    RenderResult,
} from '@testing-library/react';
import { Provider } from 'react-redux';
import { ProductItem } from "../views/shared/Products/components/ProductItem";
import {setupStore, store} from '../store/store';
import {renderWithProviders} from "../core/utils/testUtils";
import {productsData} from "../core/data/productsMockData";
const renderProductItem = (): RenderResult =>
    render(
        <Provider store={store}>
                <ProductItem addProductToCart={() => null} product={productsData[0]} />
        </Provider>
    );
describe('Product Item cases', () => {
    test('Take Snapshot', () => {
        const { asFragment } = renderProductItem();
        expect(asFragment()).toMatchSnapshot();
    });

    test('Should render products item', async () => {
        const store = setupStore();
        renderWithProviders( <ProductItem addProductToCart={() => null} product={productsData[0]} />, {store})
        const productItem = screen.getByTestId('product-item');
        const productItemName = screen.getByTestId('product-item-name');
        expect(productItem).toBeInTheDocument();
        expect(productItemName).toBeInTheDocument();
        expect(productItemName.textContent).toEqual(productsData[0].name);
    });
})

