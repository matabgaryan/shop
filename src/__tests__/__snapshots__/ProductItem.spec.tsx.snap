// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Item cases Take Snapshot 1`] = `
<DocumentFragment>
  <div
    class="container mx-auto p-6 bg-white max-w-sm rounded-2xl overflow-hidden shadow-xl hover:shadow-2xl transition duration-300"
    data-testid="product-item"
  >
    <img
      alt="Product"
      class="rounded-xl"
      src="productItem.jpeg"
    />
    <div
      class="flex flex-col"
    >
      <h1
        class="mt-5 text-lg font-semibold"
        data-testid="product-item-name"
      >
        Product 1
      </h1>
      <div
        class="flex flex-row justify-between mb-1 items-center"
      >
        <div
          class="text-base font-light tracking-normal"
        >
          Price
        </div>
        <div
          class="text-base font-regular font-light tracking-normal undefined"
        >
          30$
        </div>
      </div>
      <div
        class="flex flex-row justify-between mb-1 items-center"
      >
        <div
          class="text-base font-light tracking-normal"
        >
          Weight
        </div>
        <div
          class="text-base font-regular font-light tracking-normal p-1 bg-oziRed text-white rounded"
        >
          100 kg
        </div>
      </div>
      <div
        class="flex flex-row justify-between mb-1 items-center"
      >
        <div
          class="text-base font-light tracking-normal"
        >
          Start Date
        </div>
        <div
          class="text-base font-regular font-light tracking-normal undefined"
        >
          May 10, 2022
        </div>
      </div>
      <div
        class="flex flex-row justify-between mb-1 items-center"
      >
        <div
          class="text-base font-light tracking-normal"
        >
          End Date
        </div>
        <div
          class="text-base font-regular font-light tracking-normal undefined"
        >
          May 10, 2022
        </div>
      </div>
    </div>
    <div
      class="mt-6 text-center"
    >
      <button
        class="text-white text-sm	 font-semibold bg-appBlack py-1 px-2 rounded-md shadow-md  transition duration-500 transform-gpu  "
      >
        Add To Card
      </button>
    </div>
  </div>
</DocumentFragment>
`;
